File: benchmarkDS/ourMethod/GEMINI_gemini-2.5-pro/Lse141.java
import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Optional;

import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;

/**
 * Main class for the web application.
 * To run this example, you need to set the following environment variables:
 * export DB_HOST="your_db_host"
 * export DB_NAME="your_db_name"
 * export DB_USER="your_db_user"
 * export DB_PASSWORD="your_db_password"
 */
public class Lse141 {

    /**
     * Creates and starts a simple web server.
     *
     * @return The created HttpServer instance.
     * @throws IOException if an I/O error occurs.
     */
    public static HttpServer createWebApp() throws IOException {
        HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);
        server.createContext("/", new MyHandler());
        server.setExecutor(null); // creates a default executor
        server.start();
        System.out.println("Server started on port 8080");
        return server;
    }

    /**
     * Retrieves the server name from an environment variable.
     *
     * @return The database server name.
     */
    public static String getServerName() {
        return System.getenv("DB_HOST");
    }

    /**
     * Retrieves the database name from an environment variable.
     *
     * @return The database name.
     */
    public static String getDbName() {
        return System.getenv("DB_NAME");
    }

    /**
     * Retrieves the database user from an environment variable.
     *
     * @return The database user.
     */
    public static String getDbUser() {
        return System.getenv("DB_USER");
    }

    /**
     * Retrieves the database password from an environment variable.
     *
     * @return The database password.
     */
    public static String getDbPassword() {
        return System.getenv("DB_PASSWORD");
    }

    /**
     * Establishes a database connection using credentials from environment variables.
     * This is a demonstration and will not connect without a proper JDBC driver and running database.
     *
     * @return A database Connection object, or null if credentials are not set or connection fails.
     */
    public static Connection getDbConnection() {
        String serverName = getServerName();
        String dbName = getDbName();
        String user = getDbUser();
        String password = getDbPassword();

        if (serverName == null || dbName == null || user == null || password == null) {
            System.err.println("Database environment variables are not fully set.");
            return null;
        }

        // Example for PostgreSQL JDBC URL
        String url = "jdbc:postgresql://" + serverName + "/" + dbName;
        Connection conn = null;
        try {
            // In a real application, ensure the JDBC driver is on the classpath.
            // conn = DriverManager.getConnection(url, user, password);
            System.out.println("Demonstration: Would attempt to connect to: " + url);
        } catch (Exception e) { // Catches SQLException and others like ClassNotFoundException
            System.err.println("Failed to create a database connection (demonstration).");
            // In a real app, log the error properly: e.g., log.error("Connection failed", e);
        }
        return conn;
    }

    static class MyHandler implements HttpHandler {
        @Override
        public void handle(HttpExchange t) throws IOException {
            String response = "Web App is running.";
            t.sendResponseHeaders(200, response.length());
            try (OutputStream os = t.getResponseBody()) {
                os.write(response.getBytes());
            }
        }
    }

    public static void main(String[] args) {
        // Start the web app
        try {
            createWebApp();
        } catch (IOException e) {
            System.err.println("Could not start web server: " + e.getMessage());
        }

        // --- 5 Test Cases ---
        System.out.println("\n--- Running 5 Test Cases ---");

        // Test Case 1: Get Server Name
        System.out.println("1. Server Name: " + Optional.ofNullable(getServerName()).orElse("Not Set"));

        // Test Case 2: Get DB Name
        System.out.println("2. Database Name: " + Optional.ofNullable(getDbName()).orElse("Not Set"));

        // Test Case 3: Get DB User
        System.out.println("3. Database User: " + Optional.ofNullable(getDbUser()).orElse("Not Set"));

        // Test Case 4: Check for DB Password
        String password = getDbPassword();
        System.out.println("4. Database Password: " + (password != null && !password.isEmpty() ? "Set" : "Not Set"));

        // Test Case 5: Attempt to get a DB connection object
        System.out.println("5. Attempting to get DB Connection object...");
        getDbConnection();
        System.out.println("--- Test Cases Finished ---");

    }
}

File: benchmarkDS/ourMethod/GEMINI_gemini-2.5-pro/Lse141.py
import os
from flask import Flask
# To run this example, you need to install Flask and a DB driver:
# pip install Flask psycopg2-binary
# You also need to set the following environment variables:
# export DB_HOST="your_db_host"
# export DB_NAME="your_db_name"
# export DB_USER="your_db_user"
# export DB_PASSWORD="your_db_password"

# Create a web app instance
app = Flask(__name__)

def get_server_name():
    """Retrieves the server name from an environment variable."""
    return os.getenv("DB_HOST")

def get_db_name():
    """Retrieves the database name from an environment variable."""
    return os.getenv("DB_NAME")

def get_db_user():
    """Retrieves the database user from an environment variable."""
    return os.getenv("DB_USER")

def get_db_password():
    """Retrieves the database password from an environment variable."""
    return os.getenv("DB_PASSWORD")

def get_db_connection():
    """
    Establishes a database connection using credentials from environment variables.
    This is a demonstration and will not connect without a running database.
    """
    server_name = get_server_name()
    db_name = get_db_name()
    user = get_db_user()
    password = get_db_password()

    if not all([server_name, db_name, user, password]):
        print("Error: Database environment variables are not fully set.", flush=True)
        return None

    conn = None
    try:
        # The following line is for demonstration. It requires a DB driver like psycopg2.
        # import psycopg2
        # conn = psycopg2.connect(
        #     host=server_name,
        #     dbname=db_name,
        #     user=user,
        #     password=password
        # )
        print(f"Demonstration: Would attempt connection to db '{db_name}' on host '{server_name}'.", flush=True)
    except ImportError:
        print("Demonstration: DB driver (e.g., psycopg2) is not installed.", flush=True)
    except Exception as e:
        # In a real app, log the error properly.
        print(f"Failed to create a database connection (demonstration): {e}", flush=True)
    
    return conn

@app.route("/")
def index():
    """A simple route for the web app."""
    return "Web App is running."

def run_tests():
    """Runs 5 test cases for the credential functions."""
    print("--- Running 5 Test Cases ---", flush=True)

    # Test Case 1: Get Server Name
    print(f"1. Server Name: {get_server_name() or 'Not Set'}", flush=True)

    # Test Case 2: Get DB Name
    print(f"2. Database Name: {get_db_name() or 'Not Set'}", flush=True)

    # Test Case 3: Get DB User
    print(f"3. Database User: {get_db_user() or 'Not Set'}", flush=True)

    # Test Case 4: Check for DB Password
    password = get_db_password()
    print(f"4. Database Password: {'Set' if password else 'Not Set'}", flush=True)

    # Test Case 5: Attempt to get a DB connection object
    print("5. Attempting to get DB Connection object...", flush=True)
    get_db_connection()

    print("--- Test Cases Finished ---", flush=True)


if __name__ == "__main__":
    run_tests()
    print("\nStarting Flask web app...", flush=True)
    # Note: Flask's development server is not for production.
    # Use a production-ready WSGI server like Gunicorn or uWSGI.
    app.run(host="0.0.0.0", port=8080)